### ISSUE TITLE
Decide how alias options are handled

### ISSUE THREAD

**Issue by quintesse** [jbangdev/jbang#1080]
Aliases can supply a bunch of options that will be applied whenever an alias is run as if those options were specified on the command line by the user, eg: program arguments, java options, dependencies, system properties, etc

Aliases can also refer to other aliases.

Right now when an alias refers to another alias all its options override the options that might be set on the alias it refers to. The same also happens with options that the user might add on the command line.

So if the user runs an alias like this: `jbang --deps foo myalias` and "myalias" defines `dependencies: ["bar"]` and now imagine "myalias" refers to another alias that defines `dependencies: ["baz"]` then the final value for `dependencies` will be `["foo"]`.

Before settings this in stone for version 1.0 we might want to discuss if this is the behavior we want.

Possible implementations I can think of right now:

- **Current "override" behavior**
  - **PRO**: it's easy when dependencies are mostly used to define or override single required dependencies, eg. logger or database connection implementations
  - **CON**: you can't create aliases that add to the dependencies that have already been defined in a previous alias, you'd always have to re-state all the dependencies
- **Merge behavior**
  - **PRO**: easily add to existing dependencies
  - **CON**: unable to override dependencies already stated by another alias
- **Mixed behavior**
  - **PRO**: this would try to combine the best of both by introducing a special syntax, eg: `--deps +my.group:artifact:1.2.3` which would indicate that you don't want to override but add.
  - **CON**: more complex implementation, might be harder to understand for the user

Personally I think we can keep the current behavior because it at least doesn't make certain situations impossible while still allowing for future improvement towards the mixed behavior. Unless we think it's really important to have the mixed behavior for 1.0.

PS: I used `dependencies` as an example here, but the same holds true for `arguments`, `repositories`, `classpaths`, `java-options` and `properties`. You could theoretically imagine different strategies for each, but that might make things more confusing. The merge behavior doesn't seem to be very useful for `arguments` for example.

*Edit: decision is to keep current behavior.*

**Comment by quintesse** [comment]
It's per option indeed.

And the suggestion to put it in the key is a possibility yes, but how would we do it on the command line then? Introduce a `--+deps` flag somehow? That is probably not what you mean, right? Anyway, we can have _that_ discussion in the future.

**Comment by quintesse** [comment]
Decision to keep current behavior noted.

**Comment by maxandersen** [comment]
Hmm...ah you want it on command line too....yeah I'm not convinced we can get a nice syntax or behaviour there. I would make it override for now.

**Comment by quintesse** [comment]
Well that's what JBang already does right now, anything you put on the command line will override what the alias has. Seems only reasonable, not doing that might be really confusing ("I typed it right there, why isn't listening to me?!"). Or we'd have to spit out warning messages and that also doesn't seem a very nice solution. That's why I chose overriding back when I first implemented the aliases.
